Modify the challenge overview page

Sanji515 opened the pull request:
#### Changes proposed in this pull request:

- Changed some of the minor styling from the current version like border-radius of the card from `4px` to `10px` and similar other things also

- Link to live demo: http://pr-142-evalai.surge.sh

**Note:** Edit challenge description is not in this PR, will open another PR for that

**Screenshot:**

![Screenshot from 2019-06-10 16-58-36](https://user-images.githubusercontent.com/32524438/59192742-0ddb9280-8ba1-11e9-85ec-51eb27a66c27.png)

Sanji515 committed:
initial commit

Sanji515 committed:
Modified the challenge overview

Changed some of the minor styling from the current version
Note: Edit description feature is not in this PR.

Sanji515 committed:
fix travis errors and remove unused lines

Sanji515 committed:
minor changes

Sanji515 commented:
@RishabhJain2018 @galipremsagar @Shekharrajak @lunayach 

Please review

Sanji515 committed:
minor change

lunayach commented on src/app/components/challenge/challengeoverview/challengeoverview.component.ts:
```diff
@@ -32,5 +42,39 @@ export class ChallengeoverviewComponent implements OnInit {
+
+    const PARAMS = {
+      title: 'Edit Challenge Description',
+      content: 'asdasdasd \n\nasdasdasdasd',
```
What's this for? @Sanji515

Sanji515 replied:
@lunayach actually this will be inside the modal for editing the challenge description and have updated in the PR (#144)

putted there just to check if it is working :)

lunayach replied:
Okay. :+1:

lunayach commented:
@Sanji515 I didn't see unit tests with this PR. Do you plan to add them? (Also, I see the current build failing.)

Sanji515 commented:
@lunayach well I'm thinking to open new PR for the tests, is there anything wrong in it?

lunayach commented:
@Sanji515 It's okay, but ideally, each PR should be accompanied with the corresponding unit tests added for the added/modified code. Isn't that the purpose of unit tests? Checking (though only to some extent) working of the added code.

Sanji515 committed:
Merge branch 'master' into challenge-overview

Sanji515 committed:
resolved conflicts after updating branch with master and added media for the challenge container

Sanji515 committed:
testing on travis after removing test-log

Sanji515 committed:
testing in travis by modifying the travis.yml

Sanji515 committed:
revert travis.yml file as master

Sanji515 committed:
checking by changing privacy policy spec file

Sanji515 committed:
removed the test case for smae title and nav text in privacy policy spec file

Sanji515 committed:
removed tests for privacy policy spec

Sanji515 committed:
reverted the privacy policy spec file

Sanji515 committed:
added service in the modal spec file

Sanji515 committed:
changed the path for the servies in the spec file of modal component

Sanji515 committed:
typo in modal component spec file

Sanji515 commented:
@lunayach @Shekharrajak @galipremsagar @RishabhJain2018 

Please review

lunayach commented on src/styles/base.scss:
```diff
@@ -412,14 +503,21 @@ ul {
   margin-right: 10px;
   margin-top: 5px;
-  margin-bottom: 0 !important;
+  margin-bottom: 5px !important;
```
@Sanji515 did you check it for smaller devices?

Sanji515 replied:
@lunayach yeah I've checked I think it's looking good, but please check on your side too and let me know if any changes needed

lunayach commented on src/app/components/challenge/challenge.component.ts:
```diff
@@ -93,6 +116,147 @@ export class ChallengeComponent implements OnInit {
+          SELF.globalService.handleApiError(err, true);
+          SELF.globalService.showToast('error', err);
+        },
+        () => console.log('PUBLISH-CHALLENGE-UPDATE-FINISHED')
```
Minor: Here, it should be PUBLISH-CHALLENGE-DELETE-FINISHED
Moreover, I guess all these console logs would need to be removed when we push it to production.

Sanji515 replied:
@lunayach so should I remove all these?

lunayach replied:
I guess you could keep them for now while we/you test the application before deployment. But please don't forget to remove these afterwards.

Sanji515 replied:
@lunayach done

Sanji515 committed:
changed the console log of delete challenge and edit challenge title API

Shekharrajak commented on src/app/components/challenge/challenge.component.html:
```diff
@@ -1,42 +1,70 @@
+                                      </a>
+                                  </span>
+                                  <br>
+                                  <span style="line-height: 1.5" *ngIf="challenge.creator.team_url != ''"><strong class="text-light-black fw-semibold fs-15">Organized by:</strong><a class="orange-text" href="{{challenge['creator']['team_url']}}" target="_blank"> {{challenge['creator']['team_name']}} </a></span>
```
Try to make line legnth < 80.

Sanji515 replied:
@Shekharrajak done

Shekharrajak commented on src/app/components/challenge/challenge.component.html:
```diff
@@ -1,42 +1,70 @@
+                  <div class="row rm-row-bot card-tab-row">
+                      <div class="col-sm-12">
+                          <ul class="inline-list rm-gut">
+                              <li><a [class.selected]="localRouter.url.endsWith('overview')" [routerLink]="['./overview']" class="text-light-black fw-semibold fs-15"><i class="fa fa-info"></i> Overview</a></li>
```
You could have made it readable rather than writing everything in one line.

Sanji515 replied:
@Shekharrajak done

Shekharrajak reviewed (commented):
Can you try to write some testcases for your newly(you can try for the old one as well) created method otherwise later on no one will check all the UI if they are breaking any existing feature or functionality.

Sanji515 commented:
> Can you try to write some testcases for your newly

@Shekharrajak As other PR's dependent on each other so without merging you cannot review the functionality of other PR's, so thinking to open separate PR's for the tests as they are merged into master

Sanji515 committed:
Merge branch 'master' into challenge-overview

Shekharrajak commented:
Okay! You can open a WIP PR titled 'Specs of challenge overview page #142 ', because I don't want to forget it because it is common that developers write code and after new release old functionalities break, because of improper maintenance and specs.

Sanji515 committed:
made the line length small to make it readable

Sanji515 committed:
removed unnecessary comments in confirm component

Sanji515 commented:
@Shekharrajak I've applied the changes, please review

Sanji515 commented:
@Shekharrajak Please review

Shekharrajak commented on src/app/components/challenge/challenge.component.html:
```diff
@@ -1,42 +1,105 @@
+                                <br>
+                                <span style="line-height: 1.5" *ngIf="isChallengeHost">
+                                    <span><a class="pointer" (click)="togglePublishChallengeState()">
+                                        <strong class="text-light-black fw-semibold fs-15">{{publishChallenge.state}} </strong>
```
Are you not using any HTML prettify plugin in text editor? Plugins can do all these formattings automatically for you. Also please maintain line length < 80 (plugin is there for this as well).  HTML tags can be used line by line to make it readable.

Sanji515 replied:
@Shekharrajak how can I make HTML tags line by line for line length < 80 as it contains so many attributes inside it, instead I can wrap to be like attributes line by line

Sanji515 committed:
formatted the challenge component html

Sanji515 commented:
Hey @Shekharrajak I've formatted the code, please review if it looks good to you

Sanji515 committed:
Made the apiCall component variable

Instead of const made the apiCall a component variable so that it will be easy to write the test for the API call

Sanji515 commented:
@Shekharrajak Please review

RishabhJain2018 committed:
Merge branch 'master' into challenge-overview

RishabhJain2018 committed:
Merge branch 'master' into challenge-overview
